### PR TITLE
Forms by application filter in ES

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke pretty sure this is the only part that calls Couch for form submissions. thought about refactoring this filter for a bit, but decided against it. it's a :japanese_ogre: and would take much longer. i _think_ this is the only filter referencing form submissions in couch. all references to XFormInstance are gone in `couchaccessors`. there are still a few references to `XFormInstance` in the `reports` module, but most are in the pipeline to be changed (like exports) or benign. There's one reference in to XFormInstance in an sms report that should be trivial to tackle. There are a few more dangling CommCareCase references too

cc: @gcapalbo

### CODE REVIEW

File: corehq/apps/reports/filters/forms.py [hunk 1]
```diff
@@ -430,27 +438,12 @@ def _get_filtered_data(self, filter_results):
...
-        if parsed_params.module is not None and parsed_params.get_module_int() is None:
-            # todo: remove anytime in 2016
-            _assert(False, "module set but not a valid number!")
-            return get_form_details_for_app(domain, parsed_params.app_id, deleted=deleted)
-        elif parsed_params.most_granular_filter == 'xmlns':
-            # todo: remove anytime in 2016
-            _assert(False, "got to form ID even though this shouldn't be possible")
-            return get_form_details_for_app_and_xmlns(
-                domain, parsed_params.app_id, parsed_params.xmlns, deleted=deleted)
-        elif parsed_params.most_granular_filter == 'module':
```
Comment (benrudolph) [inline corehq/apps/reports/filters/forms.py]: @czue removed this based on your comment and testing out how the filter worked. i think this is safe, but let me know if you've seen these asserts pop up in your inbox

**Comment by snopoke** [comment]
:+1: not mergable

File: corehq/apps/reports/filters/forms.py (hunk 1 above)
Comment (czue) [inline corehq/apps/reports/filters/forms.py]: :+1: no evidence this was ever possible

**Comment by czue** [comment]
looks good! hopefully some of those changes I made in december made this a little easire.

**Comment by czue** [comment]
wow i thought we purged all these :ghost: s
```
======================================================================
FAIL: test_next_subscription (corehq.apps.accounting.tests.test_models.TestSubscription)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/accounting/tests/test_models.py", line 134, in test_next_subscription
    self.assertEqual(self.subscription.next_subscription, next_future_subscription)
AssertionError: <Subscription: Subscription to CommCare Standard Edition (v1) for DOMAIN test. [May 23, 2017 - May 31, 2017]> != <Subscription: Subscription to CommCare Standard Edition (v1) for DOMAIN test. [May 23, 2017 - Jun 24, 2017]>
----------------------------------------------------------------------
```

**Comment by benrudolph** [comment]
bah! that test has been plaguing me haha. if you were responsible for separately out the data accesses then absolutely!

**Pull request merged**